ts/ts-diabetes/src/main/java/se/inera/intyg/common/ts_diabetes/v2/testability/TSTRK1031TestabilityTestDataDecorator.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2023 Inera AB (http://www.inera.se)
+ *
+ * This file is part of sklintyg (https://github.com/sklintyg).
+ *
+ * sklintyg is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * sklintyg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package se.inera.intyg.common.ts_diabetes.v2.testability;
+
+import java.time.LocalDate;
+import se.inera.intyg.common.support.facade.util.TestabilityUtlatandeTestDataProvider;
+import se.inera.intyg.common.support.model.InternalDate;
+import se.inera.intyg.common.ts_diabetes.v2.model.internal.BedomningKorkortstyp;
+import se.inera.intyg.common.ts_diabetes.v2.model.internal.IntygAvserKategori;
+import se.inera.intyg.common.ts_diabetes.v2.model.internal.TsDiabetesUtlatandeV2;
+import se.inera.intyg.common.ts_diabetes.v2.model.internal.Vardkontakt;
+import se.inera.intyg.common.ts_parent.codes.DiabetesKod;
+import se.inera.intyg.common.ts_parent.codes.IdKontrollKod;
+
+public class TSTRK1031TestabilityTestDataDecorator implements TestabilityUtlatandeTestDataProvider<TsDiabetesUtlatandeV2> {
```

Comment:
Rename this class TSTRK1031TestabilityUtlatandeTestDataProvider. Also for consistency I would suggest calling it TsDiabetes instead of using the Tstrk...

Reply:
Done